File: tutorial/python/stdlib/containers/heap/minheap.py
#!/usr/bin/env python3
# SPDX-License-Identifier: BSD-3-Clause

import heapq


def main() -> None:
    minheap: list[int] = []
    heapq.heappush(minheap, 1)
    heapq.heappush(minheap, 12)
    heapq.heappush(minheap, 6)
    heapq.heappush(minheap, 1)
    while minheap:
        top: int = heapq.heappop(minheap)
        print(top)
    # print: 1 1 6 12


if __name__ == "__main__":
    main()

File: tutorial/java/stdlib/containers/heap/MinHeap.java
// SPDX-License-Identifier: BSD-3-Clause

import java.util.PriorityQueue;
import java.util.Queue;

public class MinHeap {
    public static void main(String[] args) {
        Queue<Integer> minheap = new PriorityQueue<>();
        minheap.add(1);
        minheap.add(12);
        minheap.add(6);
        minheap.add(1);
        while (!minheap.isEmpty()) {
            int top = minheap.poll();
            System.out.println(top);
        }
        // print: 1 1 6 12
    }
}
